compiler/lib/deadcode.ml

```diff
@@ -383,22 +384,47 @@ let remove_empty_blocks st (p : Code.program) : Code.program =
         else
           Addr.Map.add
             pc
-            { block with
-              branch =
-                (let branch = block.branch in
-                 match branch with
-                 | Branch cont -> Branch (resolve cont)
-                 | Cond (x, cont1, cont2) ->
-                     let cont1' = resolve cont1 in
-                     let cont2' = resolve cont2 in
-                     if Code.cont_equal cont1' cont2'
-                     then Branch cont1'
-                     else Cond (x, cont1', cont2')
-                 | Switch (x, a1) -> Switch (x, Array.map ~f:resolve a1)
-                 | Pushtrap (cont1, x, cont2) -> Pushtrap (resolve cont1, x, resolve cont2)
-                 | Poptrap cont -> Poptrap (resolve cont)
-                 | Return _ | Raise _ | Stop -> assert false)
-            }
+            (match block with
+            | { body; branch = Cond (x, cont1, cont2); _ } ->
+                let cont1' = resolve cont1 in
+                let cont2' = resolve cont2 in
+                if Code.cont_equal cont1' cont2'
+                then (
+                  let decr_usage x = st.live.(Var.idx x) <- st.live.(Var.idx x) - 1 in
+                  decr_usage x;
+                  let body =
+                    if List.compare_length_with body ~len:100 > 0
```

Comment:
Why such constraint?

Reply:
Large blocks are unlikely to become empty, so there is not much point in simplifying them here.
On the other hand, we are not going to process many large blocks here, so we can just as well simplify the code by removing this test.